components/sbm-recipes-boot-upgrade/src/test/java/org/springframework/sbm/boot/upgrade_27_30/ExplicitVersionTo30Test.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 - 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.sbm.boot.upgrade_27_30;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.Result;
+import org.openrewrite.maven.MavenParser;
+import org.openrewrite.maven.UpgradeDependencyVersion;
+import org.openrewrite.xml.tree.Xml;
+import org.springframework.sbm.maven.UpgradeUnmanagedSpringProject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ExplicitVersionTo30Test {
+
+    @Test
+    void shouldUpdateBomVersionTo30() {
+        Recipe recipe = new UpgradeUnmanagedSpringProject(
+                "3.0.0-M3",
+                "2\\.7\\..*");
+
+        List<Throwable> errors = new ArrayList<>();
+        InMemoryExecutionContext ctx = new InMemoryExecutionContext((ex) -> {
+            ex.printStackTrace();
```

Comment:
What about just `throw new RuntimeException(ex);` ?

Reply:
Yes, fits for test